test failures with hdf1.8.11

If the test suite is run with hdf 1.8.11, python 2.7 there are a number of errors/exceptions in `test_dimensions`  where the last character is cut off the string passed to `set_scale`.  

I suspect this a bug in hdf, as `set_scale` just calls `H5DSset_scale`, but would like a confirmation of this before I contact the hdf people.

All tests pass with hdf 1.8.8.

Example error:

```
======================================================================
FAIL: test_get_keys (_hl.tests.test_dimension_scales.TestDimensionsHighLevel)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tcaswell/other_source/h5py/build/lib.linux-x86_64-2.7/h5py/_hl/tests/test_dimension_scales.py", line 174, in test_get_keys
    self.assertEqual(self.f['data'].dims[2].keys(), ['', 'x2 name'])
AssertionError: Lists differ: [u'', u'x2 nam'] != ['', 'x2 name']

First differing element 1:
x2 nam
x2 name

- [u'', u'x2 nam']
?  -    -

+ ['', 'x2 name']
?             +
```

--- a/h5py/h5ds.pyx
+++ b/h5py/h5ds.pyx
@@ -109,7 +109,7 @@ def get_scale_name(DatasetID dscale not None):
         return ''
     name = <char*>emalloc(sizeof(char)*(namelen+1))
     try:
-        H5DSget_scale_name(dscale.id, name, namelen)
+        H5DSget_scale_name(dscale.id, name, namelen+1)
         pname = name
         return pname
     finally:

